rasterio/plot.py

```diff
@@ -33,38 +33,69 @@
 logger = logging.getLogger(__name__)
 
 
-def show(source, cmap='gray', with_bounds=True):
+def show(source, cmap='gray', with_bounds=True, ax=None, title=None, **kwargs):
     """Display a raster or raster band using matplotlib.
 
     Parameters
     ----------
-    source : array-like or (raster dataset, bidx)
+    source : array-like, or (raster dataset, bidx) tuple, or raster dataset, 
         If array-like, should be of format compatible with
         matplotlib.pyplot.imshow. If the tuple (raster dataset, bidx),
```

Comment:
Instead of trying to guess and auto-correct, let's just specify that the input array is expected in raster axis order.